Add minpac#status() function for showing status of plugins. #56

kristijanhusak opened the pull request:
Initial version of status list.
Shows only installed/updated plugins after `minpac#update()` is called.
Otherwise, returns status of all plugins.
It works similar to vim-plug, but lot of feature are missing (commit preview, handling each commit separately, etc.)
This satisfies my requirement for now, and it can be extended later on.

kristijanhusak commented:
I'm not sure why appveyor is failing.

k-takata commented:
> I'm not sure why appveyor is failing.

The appveyor environment didn't support Vim 8.1.
I have fixed it.

kristijanhusak commented:
Ok, should i rebase to make this pass?

k-takata commented on autoload/minpac/impl.vim:
```diff
@@ -427,4 +427,87 @@ function! minpac#impl#clean(args) abort
+    if !isdirectory(l:dir)
+      let l:plugin.status = 'Not installed'
+    else
+      let l:commits = systemlist([g:minpac#opt.git, '-C', l:dir, 'log',
```
I don't want to use `systemlist()` (or `system()`), because it opens a command window on Windows.
Can you use `s:system()` instead?

kristijanhusak replied:
I just tried, but that won't return any results. I tried increasing sleep to 1000m in s:system, but still i get empty results back. Do you have idea why?

kristijanhusak replied:
My bad. It works. Pushed a fix.

k-takata commented:
Can I see a screen shot of this?

kristijanhusak commented:
Sure.
Status after the minpac#update()
![update](https://i.imgur.com/L5dIGuW.png)

Calling status without minpac#update() ran before:
![status](https://i.imgur.com/Lmt9E9d.png)

pierremoreau commented:
Thank you for your work @kristijanhusak: this is something I was missing coming from vim-plug. :smiley: 

You might want to add `--no-show-signature` to the list of options given  to `git log`, as otherwise on systems where `--show-signature` is turned on by default, you end up with the following:
![currentstatus](https://user-images.githubusercontent.com/9851680/43033772-50c0d5fe-8cd0-11e8-9c8d-44a48fb247f2.png)

To display some of that signature information in a way that fits one-line outputs, you could use the `%G?` flag given to ` --pretty-format`. I am not sure if many are interested by that information, so maybe it should be behind an option that defaults to off or whatever is the value returned by `git config log.showSignature`. I do not feel strongly about adding the `%G?` flag, so feel free to disregard it if you think it is useless. Here is how it could look like, when placed between the commit hash and the subject:
![statuswithgpginfopng](https://user-images.githubusercontent.com/9851680/43033969-bfe783b2-8cd3-11e8-8756-cd4f519f6a2c.png)

kristijanhusak commented:
@pierremoreau thanks for the feedback! I guess i'll just remove the signature, it's not really important. I also plan to add a preview of commit by clicking on it in the list.

kristijanhusak commented:
@k-takata I moved some code around, hope it's fine by you.
I also added commit preview in the similar way vim-plug does it. It wasn't planned initially, but it's nice to have it.

pierremoreau commented:
Sorry for the delay, but thank you for adding the `--no-show-signature` flag, and adding the commit preview: it works very well!

kristijanhusak commented:
Great @pierremoreau, thanks for testing. @k-takata , is there something more to be done here?

k-takata commented:
I have merged this into the [devel](https://github.com/k-takata/minpac/tree/devel) branch (with some modifications).

k-takata commented:
I have released v2.0.
